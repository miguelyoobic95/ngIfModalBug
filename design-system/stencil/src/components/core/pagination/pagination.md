---
name: Pagination
category: Core Components
---
The pagination element is responsible for showing a pager object that allows browsing between pages. The pagination element is stateless. It is up to the containing element to store the current page that is being displayed and send it as a parameter 

## Attributes
|Name|Type|Default|Description|
|---|---|---|---|
|`totalItems`|number|   |The total number of items that exist and should be paginated upon|
|`itemsPerPage`|number| |Number of items that should be displayed per page|
|`currentPage`|number|0|the index of the current page being displayed. i.e. start with 0 will be page 1|
|`showTotal`|boolean|false|Should the show total be displayed. Show total is only displayed subject to screen real estate availability, if maximum pager size was reached |
|`maxPagerSize`|number| 11 |The maximum pager size to be displayed. Pager size will attempt to allocate real estate up to the maxPagerSize|

```yoo-pagination-variations.html
    <yoo-pagination total-items="100" current-page="3" items-per-page="10" class="dark"></yoo-pagination>
    <br>
    <yoo-pagination total-items="100" current-page="3" items-per-page="10" max-pager-size="5" class="dark"></yoo-pagination>
    <br>
    <yoo-pagination total-items="100" current-page="3" items-per-page="10" class="dark" show-total="true"></yoo-pagination>    
```

## CSS

|Type|Name|Description|
|---|---|---|
|Accent|`accent`|pagination accent|
|Danger|`danger`|pagination danger|
|Success|`success`|pagination success|
|Info|`info` |pagination info|
|Warning|`warning`|pagination warning|
|dark|`dark`|pagination dark|

```yoo-pagination.html
    <yoo-pagination total-items="100" current-page="3" items-per-page="10" show-total="true", max-page-size="7" class="dark"></yoo-pagination>
    <br>
    <br>
    <yoo-pagination total-items="100" current-page="3" items-per-page="10" show-total="true", max-page-size="7" class="accent"></yoo-pagination>
    <br>
    <br>
    <yoo-pagination total-items="100" current-page="3" items-per-page="10" show-total="true", max-page-size="7" class="danger"></yoo-pagination>
    <br>
    <br>
    <yoo-pagination total-items="100" current-page="3" items-per-page="10" show-total="true", max-page-size="7" class="success"></yoo-pagination>
    <br>
    <br>
    <yoo-pagination total-items="100" current-page="3" items-per-page="10" show-total="true", max-page-size="7" class="info"></yoo-pagination>
    <br>
    <br>
        <yoo-pagination total-items="100" current-page="3" items-per-page="10" show-total="true", max-page-size="7" class="warning"></yoo-pagination>
    <br>
    <br>
```

## Events

|Attr|Description|
|---|---|
|`pageChanged`|This event will be emitted when the user has clicked a request to change the page. This can be a clicking the next and prev arrows, or selecting a specific page. The event will be raised with the new page index|
|`itemsPerPageChanged`|This event will be emitted when the user has clicked a request to change the number of items per page.
