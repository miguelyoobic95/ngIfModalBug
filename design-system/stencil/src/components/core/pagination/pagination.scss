
@mixin context-container() {
    padding: 0.5rem 1rem;
    line-height: 1.5;
    width: 100%;
    position: relative;
    border-bottom: 1px solid;
}

@mixin color-pagination($color, $color-20, $color-05) {
    .container {
        .inner-container {
            .page-number {
                color: $color;
                font-size: 0.8rem;
                font-weight: 400;
                &:hover {
                    background: $color-20;
                }
                &.active {
                    background: $color;
                    color: cssvar(light);
                }
                &.disabled {
                    color: $color-05;
                    &:hover {
                        cursor: not-allowed;
                        background: transparent;
                    }
                }
                &.ellipsis {
                    &:hover{
                        cursor: default;
                        background: transparent;
                    }
                }
            }
        }
    }  
}

:host {
    .container {
        width: 100%;
        &.show-total-disabled {
            justify-content: center;
        }
         &.show-total {
            justify-content: space-between;
        }
        .total-container {
            color: cssvar(dark-80);
            align-items: center;
            min-width: 7.6rem;
            .text-container {
                padding-left: 0.3rem;
            }
            .dropdown-entry {
                padding: 0.5rem 1rem;
                line-height: 1.5;
                width: 100%;
                cursor: pointer;
                &:hover {
                    background: cssvar(accent-05);
                    color: cssvar(dark);
                }
            }
            .items-container {
                padding-left: 0.5rem;
            }
        }
        .inner-container {
            .page-number {
                align-items: center;
                justify-content: center;
                border-radius: 5px;
                height: 2.5rem;
                width: 2.5rem;
                &:hover {
                    cursor: pointer;
                }
            }
        }
    }
}

:host(.accent) {
    @include color-pagination(cssvar(accent), cssvar(accent-20), cssvar(accent-05));
}

:host(.danger) {
    @include color-pagination(cssvar(danger), cssvar(danger-20), cssvar(danger-05));
}

:host(.success) {
     @include color-pagination(cssvar(success), cssvar(success-20), cssvar(success-05));
}

:host(.info) {
    @include color-pagination(cssvar(info), cssvar(info-20), cssvar(info-05));
}

:host(.warning) {
     @include color-pagination(cssvar(warning), cssvar(warning-20), cssvar(warning-05));
}

:host(.dark) {
    @include color-pagination(cssvar(dark-80), cssvar(dark-40), cssvar(dark-20));
}