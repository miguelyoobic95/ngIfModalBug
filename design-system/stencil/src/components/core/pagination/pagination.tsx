import { Component, Prop, Event, EventEmitter, State, Element } from '@stencil/core';
import { getElementDimensions } from '../../../utils/helpers';

const ITEM_WIDTH: number = 40;
const ARROWS_SIZE: number = ITEM_WIDTH * 2;
const DESCRIPTION_LENGTH: number = 310; // need to adjust for translation
const TRUNCATION = '…';
const MIN_TRUNCATION_PAGER = 5;
@Component({
    tag: 'yoo-pagination',
    styleUrl: 'pagination.scss',
    scoped: true
})

export class YooPaginationComponent {

    @Prop() totalItems: number;
    @Prop() itemsPerPage: number;
    @Prop() currentPage: number = 0; // index number, i.e currentPage=0 will be page 1
    @Prop() showTotal: boolean;
    @Prop() maxPagerSize: number = 11;

    @Event({ bubbles: true, composed: true }) pageChanged: EventEmitter<any>;
    @Event() itemsPerPageChanged: EventEmitter<number>;

    @State() pagerSize: number; // including ellipsis
    @State() totalVisible: boolean;

     @Element() host: HTMLStencilElement;

    get currentPageDisplay() {
        return this.currentPage + 1;
    }

    get totalPages() {
        return this.itemsPerPage > 0 ? Math.ceil(this.totalItems / this.itemsPerPage) : 0;
    }

    isLastPage(page) {
        return page >= this.totalPages;
    }

    isFirstPage(page) {
        return page === 1;
    }

    hasNext() {
        return !this.isLastPage(this.currentPageDisplay);
    }

    hasPrevious() {
        return !this.isFirstPage(this.currentPageDisplay);
    }

    isTotalVisible(width): boolean {
        const minWidthWithDescription: number = DESCRIPTION_LENGTH + ARROWS_SIZE + this.getPagerSize(width) * ITEM_WIDTH;
        return this.showTotal && width > minWidthWithDescription;
    }

    getPagerSize(width: number): number {
        width -= ARROWS_SIZE;
        let availableItems = Math.floor(width / ITEM_WIDTH);
        return Math.min(this.totalPages, Math.max(0, availableItems), this.maxPagerSize);
    }

    getDisplayValue(position: number): string | number {
        // no truncation required. position = page number.
        if (this.totalPages <= this.pagerSize) {
            return position;
        }
        let leftTruncationPos = this.currentPageDisplay >= this.pagerSize - 1 ? 2 : undefined;
        let rightTruncationPos = this.totalPages - this.currentPageDisplay + 1 >= this.pagerSize - 1 ? this.pagerSize - 1 : undefined;

        let currentPagePosition = Math.ceil(this.pagerSize / 2);

        if (this.pagerSize < MIN_TRUNCATION_PAGER) {
            if (leftTruncationPos && !rightTruncationPos) {
                currentPagePosition = this.pagerSize - this.totalPages + this.currentPageDisplay;
            }
            if (!leftTruncationPos && rightTruncationPos) {
                currentPagePosition = this.currentPageDisplay;
            }
            return this.currentPageDisplay - currentPagePosition + position;
        }

        // Both truncations
        if (rightTruncationPos && leftTruncationPos) {
            if (position === leftTruncationPos || position === rightTruncationPos) {
                return TRUNCATION;
            }
            if (position === 1) {
                return position;
            }
            if (position === this.pagerSize) {
                return this.totalPages;
            }
            return this.currentPageDisplay - currentPagePosition + position;
        }
        // only right or left truncation
        let truncationPosition = leftTruncationPos || rightTruncationPos;
        if (position < truncationPosition) { return position; }
        if (position > truncationPosition) { return this.totalPages - this.pagerSize + position; }
        return TRUNCATION;
    }

    moveToPage(page: number | string): void {
        let newPage = this.currentPageDisplay;
        if (page === 'right' && this.hasNext()) {
            newPage++;
        }
        if (page === 'left' && this.hasPrevious()) {
            newPage--;
        }
        if (typeof page === 'number' && page >= 1 && page <= this.totalPages) {
            newPage = page;
        }
        if (this.pageChanged && this.currentPageDisplay !== newPage) {
            this.pageChanged.emit(newPage - 1);
        }
    }

    updateVisibility(): void {
        let width = getElementDimensions(this.host).width;
        this.pagerSize = this.getPagerSize(width);
        this.totalVisible = this.isTotalVisible(width);
    }

    updateItemsPerPage(newItemsPerPage: number) {
        this.itemsPerPageChanged.emit(newItemsPerPage);
    }

    componentDidLoad(): void {
        this.updateVisibility();
        parent.addEventListener('resize', () => this.updateVisibility()); //This implementaion must be used otherwise the host element will become undefined on page resize.
    }

    componentDidUpdate() {
        this.updateVisibility();
    }

    // rendering functions
    renderTotalDescription(): JSX.Element {
        if (this.totalVisible) {
            return (
                <div class="total-container" attr-layout="row">
                    <span class="text-container">Number of items per page:</span>
                    <yoo-context-menu>
                            <div slot="trigger" class="text-container">
                                {' ' + this.itemsPerPage + ' '}<span class="yo-arrow-dropdown"></span>
                            </div>
                            <div class="dropdown-entry" onClick={() => this.updateItemsPerPage(10)}>10</div>
                            <div class="dropdown-entry" onClick={() => this.updateItemsPerPage(25)}>25</div>
                            <div class="dropdown-entry" onClick={() => this.updateItemsPerPage(100)}>100</div>
                            <div class="dropdown-entry" onClick={() => this.updateItemsPerPage(500)}>500</div>
                    </yoo-context-menu>
                    <span class="items-container" slot="trigger" attr-layout="row"> Total items: {this.totalItems}</span>
                </div>
            );
        }
        return null;
    }

    renderPageItem(pos: number): JSX.Element {
        let page = this.getDisplayValue(pos);
        if (page === '…') {
            return (
                <span class="page-number ellipsis" attr-layout="row">
                    <span class="yo-more"></span>
                </span>
            );
        }
        return (
            <span class={'page-number ' + (page === this.currentPageDisplay ? 'active' : '')}
                onClick={() => this.moveToPage(page)} attr-layout="row">
                {page}
            </span>
        );
    }

    render() {

        return (
            <div attr-layout="row" class={'container ' + (this.totalVisible ? 'show-total' : 'show-total-disabled')}>
                {this.renderTotalDescription()}
                <div attr-layout="row" class="inner-container">
                    <div class={'page-number ' + (!this.hasPrevious() ? 'disabled' : '')} attr-layout="row" onClick={this.hasPrevious() ? () => this.moveToPage('left') : null}>
                        <span class="yo-left"></span>
                    </div>
                    {[...Array(this.pagerSize)].fill(0).map((_, k) => {
                        return this.renderPageItem(k + 1);
                    })}

                    <div class={'page-number ' + (!this.hasNext() ? 'disabled' : '')} attr-layout="row" onClick={this.hasNext() ? () => this.moveToPage('right') : null}>
                        <span class="yo-right"></span>
                    </div>
                </div>

            </div>
        );
    }
}
