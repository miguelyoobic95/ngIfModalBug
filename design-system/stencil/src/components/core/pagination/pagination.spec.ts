import { TestWindow } from '@stencil/core/testing';

let mockDimensions = jest.fn();
jest.mock('../../../utils/helpers', () => {
    return {
        getElementDimensions: mockDimensions
    };
});

import { YooPaginationComponent } from './pagination';

describe('YooPaginationComponent', () => {
    let window;
    beforeEach(() => {
      window = new TestWindow();
    });
    it('should instantiate', () => {
        expect(new YooPaginationComponent()).toBeTruthy();
    });
    describe ('Total Pages calculations', () => {

        let comp;
        beforeEach(() => {
            comp = new YooPaginationComponent();
        });

        it('should return extra page for extra items', () => {
            comp.totalItems = 16;
            comp.itemsPerPage = 10;
            expect(comp.totalPages).toEqual(2);
        });

        it('should return exact number of total pages', () => {
            comp.totalItems = 10;
            comp.itemsPerPage = 10;
            expect(comp.totalPages).toEqual(1);
        });

        it('should return 0 if items per page is 0 ', () => {
            comp.totalItems = 16;
            comp.itemsPerPage = 0;
            expect(comp.totalPages).toEqual(0);
        });

        it('should return 0 if items per page is not defined ', () => {
            comp.totalItems = 16;
            expect(comp.totalPages).toEqual(0);
        });

    });

    describe('calcualte pager size', () => {

        let comp;
        beforeEach(() => {
            comp = new YooPaginationComponent();
        });

        it('should return total items. No description', () => {
            comp.totalItems = 10;
            comp.itemsPerPage = 10;
            expect(comp.getPagerSize(1000)).toEqual(1);
        });

        it('should return total items. With description', () => {
            comp.showTotal = true;
            comp.totalItems = 10;
            comp.itemsPerPage = 10;
            expect(comp.getPagerSize(1000)).toEqual(1);
        });

        it('should return space available. No description', () => {
            comp.showTotal = false;
            comp.totalItems = 100;
            comp.itemsPerPage = 10;
            expect(comp.getPagerSize(80 + 125)).toEqual(3);
        });

        it('should return space available. Description omitted', () => {
            comp.showTotal = true;
            comp.totalItems = 100;
            comp.itemsPerPage = 10;
            expect(comp.getPagerSize(80 + 140 + 125)).toEqual(6);
        });

        it('should return max pager size', () => {
            comp.showTotal = true;
            comp.totalItems = 200;
            comp.itemsPerPage = 10;
            expect(comp.getPagerSize(1000)).toEqual(11);
        });

        it('should return provided max pager size', () => {
            comp.showTotal = true;
            comp.totalItems = 100;
            comp.maxPagerSize = 8;
            comp.itemsPerPage = 10;
            expect(comp.getPagerSize(1000)).toEqual(8);
        });

        it('should return 0 when width is 0', () => {
            comp.showTotal = true;
            comp.totalItems = 100;
            comp.maxPagerSize = 8;
            comp.itemsPerPage = 10;
            expect(comp.getPagerSize(0)).toEqual(0);
        });

    });

    describe('calculate display values', () => {

        let comp;
        function calculateValues (size: number): Array<string | number>  {
            return [...Array(size)].fill(0).map( (_, k) => comp.getDisplayValue(k + 1) );
        }

        let mockTotalPages;
        beforeEach(() => {
            mockTotalPages = jest.fn();
            comp = new YooPaginationComponent();
            Object.defineProperty(comp, 'totalPages', {
                get: mockTotalPages,
                configurable: true
            });
        });

        it ('should display all pages', () => {
            const pagerSize = 5;
            comp.pagerSize = pagerSize;
            comp.currentPage = 1;
            mockTotalPages.mockImplementation(() => pagerSize);
            let res = calculateValues(pagerSize);
            expect(res).toEqual([1, 2, 3, 4, 5]);
        });

        describe ('odd array', () => {
            const pagerSize = 7;
            let res: Array<string | number> = [];
            beforeEach (() => {
                comp.pagerSize = pagerSize;
                mockTotalPages.mockImplementation(() => 100);
            });
            const results = [
                { current: 95, values: [1, '…', 96, 97, 98, 99, 100]},
                { current: 99, values: [1, '…', 96, 97, 98, 99, 100]},
                { current: 0, values: [1, 2, 3, 4, 5, '…', 100]},
                { current: 4, values: [1, 2, 3, 4, 5, '…', 100]},
                { current: 49, values: [1, '…', 49 , 50 , 51 , '…', 100]}
            ];
            results.forEach(r => {
                it('should display values when current page is ' + r.current, () => {
                    comp.currentPage = r.current;
                    res = calculateValues(pagerSize);
                    expect(res).toEqual(r.values);
                });
            });
        });

        describe ('even array', () => {
            const pagerSize = 6;
            let res: Array<string | number> = [];
            beforeEach (() => {
                comp.pagerSize = pagerSize;
                mockTotalPages.mockImplementation(() => 100);
            });
            const results = [
                { current: 96, values: [1, '…', 97, 98, 99, 100]},
                { current: 99, values: [1, '…', 97, 98, 99, 100]},
                { current: 0, values: [1, 2, 3, 4, '…', 100]},
                { current: 3, values: [1, 2, 3, 4, '…', 100]},
                { current: 49, values: [1, '…', 50 , 51 , '…', 100]}
            ];
            results.forEach(r => {
                it('should display values when current page is ' + r.current, () => {
                    comp.currentPage = r.current;
                    res = calculateValues(pagerSize);
                    expect(res).toEqual(r.values);
                });
            });
        });

        describe ('small array', () => {
            const pagerSize = 3;
            let res: Array<string | number> = [];
            beforeEach (() => {
                comp.pagerSize = pagerSize;
                mockTotalPages.mockImplementation(() => 100);
            });
            const results = [
                { current: 98, values: [98, 99, 100]},
                { current: 99, values: [98, 99, 100]},
                { current: 0, values: [1, 2, 3]},
                { current: 1, values: [1, 2, 3]},
                { current: 49, values: [49 , 50 , 51]}
            ];
            results.forEach(r => {
                it('should display values when current page is ' + r.current, () => {
                    comp.currentPage = r.current;
                    res = calculateValues(pagerSize);
                    expect(res).toEqual(r.values);
                });
            });
        });

    });

    describe('Snapshots', () => {

        it ('should have page numbers and active pages', async () => {
            mockDimensions.mockImplementation(() => {
                return {width: 500, height: 100};
            });
            let element = await window.load({
                components: [YooPaginationComponent],
                html: '<yoo-pagination total-items="100" items-per-page="10" current-page="3" class="dark"></yoo-pagination></yoo-pagination>'
            });
            expect(element).toMatchSnapshot();
        });

        it ('should have small pager', async () => {
            mockDimensions.mockImplementation(() => {
                return {width: 150, height: 100};
            });
            let element = await window.load({
                components: [YooPaginationComponent],
                html: '<yoo-pagination total-items="100" items-per-page="10" current-page="3" show-total="true" class="dark"></yoo-pagination></yoo-pagination>'
            });
            expect(element).toMatchSnapshot();
        });

        it ('should have total shown', async () => {
            mockDimensions.mockImplementation(() => {
                return {width: 1000, height: 100};
            });
            let element = await window.load({
                components: [YooPaginationComponent],
                html: '<yoo-pagination total-items="50" items-per-page="10" current-page="3" class="dark"></yoo-pagination></yoo-pagination>'
            });
            expect(element).toMatchSnapshot();
        });

    });

  describe('events', () => {
    let element, res, emitEvent;
    beforeEach( async () => {
      mockDimensions.mockImplementation(() => {
        return {width: 500, height: 100};
      });
      window = new TestWindow();
      emitEvent = jest.fn(ev => { res = ev.detail; });
    });
    it ('should emit pageChanged on click', async () => {
      element = await window.load({
        components: [YooPaginationComponent],
        html: ` <yoo-pagination total-items="100" items-per-page="10" current-page="0" class="dark"></yoo-pagination>`
      });
      window.document.addEventListener('pageChanged', emitEvent);
      let activePage = element.querySelector('.active');
      expect(activePage.textContent).toEqual('1');
      let moveRight = element.querySelector('.yo-right');
      await moveRight.click();
      expect(emitEvent).toHaveBeenCalled();
      expect(res).toEqual(1);
    });
    it ('should emit itemsPerPageChanged on click', async () => {
      element = new YooPaginationComponent();
      element.itemsPerPageChanged = {
          emit: () => {}
      };
      const changedSpy = jest.spyOn(element.itemsPerPageChanged, 'emit');
      element.updateItemsPerPage(24);
      expect(changedSpy).toHaveBeenCalledWith(24);
    });
  });

});
