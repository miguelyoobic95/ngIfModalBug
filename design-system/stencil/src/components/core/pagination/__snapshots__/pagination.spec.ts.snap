// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`YooPaginationComponent Snapshots should have page numbers and active pages 1`] = `
<yoo-pagination
  class="dark hydrated"
  current-page="3"
  data-yoo-pagination-host=""
  items-per-page="10"
  total-items="100"
>
  
  <div
    attr-layout="row"
    class="container show-total-disabled"
    data-yoo-pagination=""
  >
    
    <div
      attr-layout="row"
      class="inner-container"
      data-yoo-pagination=""
    >
      <div
        attr-layout="row"
        class="page-number"
        data-yoo-pagination=""
      >
        <span
          class="yo-left"
          data-yoo-pagination=""
        />
      </div>
      <span
        attr-layout="row"
        class="page-number"
        data-yoo-pagination=""
      >
        1
      </span>
      <span
        attr-layout="row"
        class="page-number"
        data-yoo-pagination=""
      >
        2
      </span>
      <span
        attr-layout="row"
        class="page-number"
        data-yoo-pagination=""
      >
        3
      </span>
      <span
        attr-layout="row"
        class="page-number active"
        data-yoo-pagination=""
      >
        4
      </span>
      <span
        attr-layout="row"
        class="page-number"
        data-yoo-pagination=""
      >
        5
      </span>
      <span
        attr-layout="row"
        class="page-number"
        data-yoo-pagination=""
      >
        6
      </span>
      <span
        attr-layout="row"
        class="page-number"
        data-yoo-pagination=""
      >
        7
      </span>
      <span
        attr-layout="row"
        class="page-number"
        data-yoo-pagination=""
      >
        8
      </span>
      <span
        attr-layout="row"
        class="page-number"
        data-yoo-pagination=""
      >
        9
      </span>
      <span
        attr-layout="row"
        class="page-number"
        data-yoo-pagination=""
      >
        10
      </span>
      <div
        attr-layout="row"
        class="page-number"
        data-yoo-pagination=""
      >
        <span
          class="yo-right"
          data-yoo-pagination=""
        />
      </div>
    </div>
  </div>
</yoo-pagination>
`;

exports[`YooPaginationComponent Snapshots should have small pager 1`] = `
<yoo-pagination
  class="dark hydrated"
  current-page="3"
  data-yoo-pagination-host=""
  items-per-page="10"
  show-total="true"
  total-items="100"
>
  
  <div
    attr-layout="row"
    class="container show-total-disabled"
    data-yoo-pagination=""
  >
    
    <div
      attr-layout="row"
      class="inner-container"
      data-yoo-pagination=""
    >
      <div
        attr-layout="row"
        class="page-number"
        data-yoo-pagination=""
      >
        <span
          class="yo-left"
          data-yoo-pagination=""
        />
      </div>
      <span
        attr-layout="row"
        class="page-number"
        data-yoo-pagination=""
      >
        10
      </span>
      <div
        attr-layout="row"
        class="page-number"
        data-yoo-pagination=""
      >
        <span
          class="yo-right"
          data-yoo-pagination=""
        />
      </div>
    </div>
  </div>
</yoo-pagination>
`;

exports[`YooPaginationComponent Snapshots should have total shown 1`] = `
<yoo-pagination
  class="dark hydrated"
  current-page="3"
  data-yoo-pagination-host=""
  items-per-page="10"
  total-items="50"
>
  
  <div
    attr-layout="row"
    class="container show-total-disabled"
    data-yoo-pagination=""
  >
    
    <div
      attr-layout="row"
      class="inner-container"
      data-yoo-pagination=""
    >
      <div
        attr-layout="row"
        class="page-number"
        data-yoo-pagination=""
      >
        <span
          class="yo-left"
          data-yoo-pagination=""
        />
      </div>
      <span
        attr-layout="row"
        class="page-number"
        data-yoo-pagination=""
      >
        1
      </span>
      <span
        attr-layout="row"
        class="page-number"
        data-yoo-pagination=""
      >
        2
      </span>
      <span
        attr-layout="row"
        class="page-number"
        data-yoo-pagination=""
      >
        3
      </span>
      <span
        attr-layout="row"
        class="page-number active"
        data-yoo-pagination=""
      >
        4
      </span>
      <span
        attr-layout="row"
        class="page-number"
        data-yoo-pagination=""
      >
        5
      </span>
      <div
        attr-layout="row"
        class="page-number"
        data-yoo-pagination=""
      >
        <span
          class="yo-right"
          data-yoo-pagination=""
        />
      </div>
    </div>
  </div>
</yoo-pagination>
`;
